src/nft/hooks/usePermit2Approval.ts

```diff
@@ -17,14 +18,22 @@ export default function usePermit2Approval(
   shouldUseNftRouter?: boolean
 ) {
   const { chainId } = useWeb3React()
+  const shouldUseUpdatedContract = usePwatNewContractEnabled()
+
+  const canPermit = enabled && chainId
+  let contractToApprove: string | undefined
+
+  if (canPermit) {
+    const forceNftRouter = shouldUseNftRouter && chainId === 1
+    const mainnetContractToApprove = shouldUseUpdatedContract
+      ? NFT_UNIVERSAL_ROUTER_MAINNET_ADDRESS
+      : DEPRECATED_NFT_UNIVERSAL_ROUTER_MAINNET_ADDRESS
+    contractToApprove = forceNftRouter ? mainnetContractToApprove : UNIVERSAL_ROUTER_ADDRESS(chainId)
+  }
```

Comment:
This is pretty confusing with all the ternaries.
Can you move this to an unexported hook that just returns the contract, so you can use if/else statements instead of ternaries?